src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_field.ts

```diff
@@ -0,0 +1,378 @@
+          // First filter out the validation type we are currently validating
+          const filteredErrors = filterErrors(previousErrors, validationType);
+          return [...filteredErrors, ...validationErrors];
+        });
+      }
+      return {
+        isValid: validationErrors.length === 0,
+        errors: validationErrors,
+      };
+    };
+
+    if (isValidationAsync) {
+      return validateAsync({
+        formData,
+        value: valueToValidate,
+        validationTypeToValidate: validationType,
+      }).then(onValidationResult);
+    } else {
+      const validationErrors = validateSync({
+        formData,
+        value: valueToValidate,
+        validationTypeToValidate: validationType,
+      });
+      return onValidationResult(validationErrors);
+    }
+  };
+
+  /**
+   * Handler to change the field value
+   *
+   * @param newValue The new value to assign to the field
+   */
+  const setValue: Field['setValue'] = newValue => {
+    onValueChange();
+
+    const formattedValue = runFormatters(newValue);
+    setStateValue(formattedValue);
+
+    // Update the form data observable
+    form.__updateFormDataAt(path, getOutputValue(formattedValue));
+  };
+
+  /**
+   * Form <input /> "onChange" event handler
+   *
+   * @param event Form input change event
+   */
+  const onChange: Field['onChange'] = event => {
+    const newValue = {}.hasOwnProperty.call(event!.target, 'checked')
+      ? event.target.checked
+      : event.target.value;
+
+    setValue(newValue);
+  };
+
+  /**
+   * As we can have multiple validation types (FIELD, ASYNC, ARRAY_ITEM), this
+   * method allows us to retrieve error messages for certain types of validation.
+   *
+   * For example, if we want to validation error messages to be displayed when the user clicks the "save" button
+   * _but_ in case of an asynchronous validation (for ex. an HTTP request that would validate an index name) we
+   * want to immediately display the error message, we would have 2 types of validation: FIELD & ASYNC
+   *
+   * @param validationType The validation type to return error messages from
+   */
+  const getErrorsMessages: Field['getErrorsMessages'] = (
+    validationType = VALIDATION_TYPES.FIELD
+  ) => {
+    const errorMessages = errors.reduce((messages, error) => {
+      if (
+        error.validationType === validationType ||
+        (validationType === VALIDATION_TYPES.FIELD &&
+          !{}.hasOwnProperty.call(error, 'validationType'))
+      ) {
+        return messages ? `${messages}, ${error.message}` : (error.message as string);
+      }
+      return messages;
+    }, '');
+
+    return errorMessages ? errorMessages : null;
+  };
+
+  const getOutputValue: Field['__getOutputValue'] = (rawValue = value) => serializer(rawValue);
+
+  // -- EFFECTS
+  // ----------------------------------
+  useEffect(() => {
+    if (isPristine) {
+      // Avoid validate on mount
+      return;
+    }
+    form.__validateFields(fieldsToValidateOnChange);
```

Comment:
Good catch, although in practice I doubt we would miss the fact that a field is not being validated (when manually testing the validation). What you suggest would trigger twice the validation in case the `path` is already included in `fieldsToValidateOnChange`.

We could add a check at the top of the hook to verify that the `path` is indeed in the array but I prefer to wait to feel the pain before adding this maybe unnecessary check.